app/controllers/materials_controller.rb

```diff
@@ -0,0 +1,221 @@
+require 'rubygems'
```

Comment:
Most of it consists in checking if the parameters are valid, so we figured that's a task for the controller though. Open to suggestions, though, what parts do you think should be moved to another class?